### DISCUSSION TITLE
MD5 hashing algorithm

### DISCUSSION THREAD

**Discussion by ar-comlog** [drakkan/sftpgo#780] [General]
I want to transfer settings from FileZilla and Silvershild to SFTPgo. The passwords are stored in MD5 format. It would be cool if you could specify md5 instead of bcrypt in the configuration.

**Comment by drakkan** [comment]
If your passwords are something like these:

- `$1$01234567$b5lh2mHyD2PdJjFfALlEz1`
- `$apr1$01234567$IXBaQywhAhc0d75ZbaSDp/`

they will work, otherwise you are out of luck, md5 is too old and insecure, I don't think I'll add support for it, sorry

**Reply by drakkan** [reply]
This is now supported, see https://github.com/drakkan/sftpgo/commit/5a40f998aef2ca4a52b8f990ee6cccdf7f4000aa

If your password is `5f4dcc3b5aa765d61d8327deb882cf99`, you can import in the SFTPGo data provider as `{MD5}5f4dcc3b5aa765d61d8327deb882cf99`. It will be converted to `bcrypt` on user login

**Reply by ar-comlog** [reply]
Thank you very much! My password format is 95757bfa00bf85b17921d709b5af0232. I am happy with the solution {MD5}95757bfa00bf85b17921d709b5af0232!
